pygbm/loss.py

```diff
@@ -0,0 +1,16 @@
+import numpy as np
+
+
+# TODO: ABC for losses, regression losses and classification losses
+class LeastSquares:
+    def __init__(self):
+        self.hessians = np.full(shape=1, fill_value=2, dtype=np.float32)
+
```

Comment:
Please add a `__call__` method that computes the actual loss value and use it in tests to make sure that the derivatives agree with the loss: e.g. that a minimum is reached when taking 1d newton steps from some arbitrary init: try with `y_true = [42.]`, `y_pred_init = [-12]` and use:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.newton.html

to iteratively update `y_pred` until reaching convergence and check that the loss value at convergence is 0 (e.g. for least squares regression and for classification with the binomial deviance, that is with `y_true` with values in `{0, 1}`).

Reply:
You can also try write tests with 5D `y_true` / `y_pred` arrays: small dimensions that makes it easy to write test with hardcoded values in the test but not as trivial as the 1D case.